apps/topics-mine/src/python/topics_mine/posts/pushshift_twitter.py

```diff
@@ -0,0 +1,291 @@
+        
+        return {
+            'size': PS_TWITTER_PAGE_SIZE,
+            'sort': ['_doc'],
+            'query': {'ids': {'values': ids}}
+        }
+
+    def _build_range_query(self, query: dict, start_date: datetime, end_date: datetime) -> dict:
+        """Build an ElasticSearch query filtered by date range."""
+        
+        filters = [{'match': {'text': query}}]
+        if start_date:
+            start_date_ts = int(start_date.timestamp())
+            filters.append({'range': {'created_at': {'gte': start_date_ts}}})
+        if end_date:
+            end_date_ts = int(end_date.timestamp())
+            filters.append({'range': {'created_at': {'lt': end_date_ts}}})
+
+        prepared_query = {
+            'size': PS_TWITTER_PAGE_SIZE,
+            'sort': ['_doc'],
+            'query': {'bool': {'must': filters}}
+        }
+        
+        return prepared_query
+
+    def _extract_quote_tweets(self, tweets: list) -> dict:
+        """Find quote tweets then map IDs of the original tweet to lists of their quote tweets.
+        
+        Returns: {original_tweet_id:[quote_tweet_dict]}
+        """
+
+        quote_tweets = {}
+        for tweet in tweets:
+            if tweet.get('is_quote_status'):
+                quoted_status_id = tweet.get('quoted_status_id')
+                if quoted_status_id:
+                    quote_tweets.setdefault(quoted_status_id, []).append(tweet)
+                else:
+                    log.warning('quote tweet missing quoted_status_id for id %d' % tweet['id'])
+        
+        return quote_tweets
+
+    def _extract_retweets(self, tweets: list) -> dict:
+        """Find retweets then map IDs of the original tweet to lists of their retweets.
+        
+        Returns: {original_tweet_id:[retweet_dict]}
+        """
+
+        retweets = {}
+        for tweet in tweets:
+            if tweet.get('is_retweet_status'):
+                retweeted_status_id = tweet.get('retweeted_status_id')
+                if retweeted_status_id:
+                    retweets.setdefault(retweeted_status_id, []).append(tweet)
+                else:
+                    log.warning('retweet missing retweeted_status_id for id %d' % tweet['id'])
+        
+        return retweets
+
+    def _fetch_referenced_tweets(self, tweets: list) -> None:
+        """Fetch quote tweets and retweets referenced by tweets in the list.
+
+        Since Pushshift doesn't store the referenced tweet dict for quote tweets or retweets, only
+        their IDs, a second roundtrip is necessary to complete records with a referenced tweet.
+        
+        For example, on a quote tweet the quoted_status field isn't present, only quoted_status_id.
+        Similarly, on a retweet the retweeted_status field isn't present, only retweeted_status_id.
+
+        Returns: None, referenced tweets are stored in tweet['quoted_status'] or
+        tweet['retweeted_status'] as appropriate.
```

Comment:
So this modifies the parameter `tweets` in-place, right?

Could you make it return the modified tweets instead? Now it's a bit hard to figure out which specific method does add those `quoted_status` / `retweeted_status` to the `tweets`. We have a plenty of those C-style "pass a reference" methods that change up their arguments in subtle ways all over our codebase, and I can't say I like them too much :)